azure/sdk-identity-resources-storage/Program.cs

```diff
@@ -0,0 +1,137 @@
+
+            DefaultAzureCredential credentials = new DefaultAzureCredential();
+
+            ResourcesManagementClient resourcesManagementClient = new ResourcesManagementClient(subscriptionId, credentials);
+            StorageManagementClient storageManagementClient = new StorageManagementClient(subscriptionId, credentials);
+
+            // Create a Resource Group
+            string resourceGroupName = await CreateResourceGroup(resourcesManagementClient);
+
+            // Create a Storage account
+            string storageName = await CreateStorageAccount(storageManagementClient, resourceGroupName);
+
+            // Create a container and upload a blob
+            await UploadBlobToStorageAccountUsingClientConnectionString(storageManagementClient, resourceGroupName, storageName);
+
+            Console.WriteLine("Press any key to continue and delete the resources...");
+            Console.ReadKey(true);
+
+            await DeleteResourceGroup(resourcesManagementClient, resourceGroupName);
+
+            return;
+        }
+
+        private static async Task DeleteResourceGroup(ResourcesManagementClient resourcesManagementClient, string resourceGroupName)
+        {
+            Console.WriteLine($"Deleting resource group {resourceGroupName}...");
+            ResourceGroupsDeleteOperation deleteOperation = await resourcesManagementClient.ResourceGroups.StartDeleteAsync(resourceGroupName);
+            await deleteOperation.WaitForCompletionAsync();
+            Console.WriteLine("Done!");
+        }
+
+        private static async Task<string> CreateStorageAccount(StorageManagementClient storageManagementClient, string rgName)
+        {
+            Console.WriteLine("Creating a new storage account...");
+
+            string storageAccountName = await GetStorageAccountName(storageManagementClient);
+            StorageAccountCreateParameters parms = new StorageAccountCreateParameters(
+                        new Sku("Standard_LRS"),
+                        Kind.BlobStorage,
+                        "West US");
+            parms.AccessTier = AccessTier.Hot;
+
+            StorageAccountsCreateOperation createStorageAccount =
+                await storageManagementClient.StorageAccounts.StartCreateAsync(rgName, storageAccountName, parms);
+                        await createStorageAccount.WaitForCompletionAsync();
+            Console.WriteLine($"Done creating account {storageAccountName}.");
+
+            return storageAccountName;
+        }
+
+        private static async Task<string> GetStorageAccountName(StorageManagementClient storageManagementClient)
+        {
+            bool foundAvailableName = false;
+            string storageAccountName = string.Empty;
+
+            while (!foundAvailableName)
+            {
+                storageAccountName = RandomName("storage", 20);
+                Response<CheckNameAvailabilityResult> availability =
```

Comment:
Nit - call this `availabilityResponse` so it's less weird when you unwrap the `.Value`.